logstash-core/lib/logstash/pipeline.rb

```diff
@@ -250,12 +250,32 @@ def stopped?
     @running.false?
   end
 
+  # register_plugin simply calls the plugin #register method and catches & logs any error
+  # @param plugin [Plugin] the plugin to register
+  # @return [Plugin] the registered plugin
+  def register_plugin(plugin)
+    plugin.register
+    plugin
+  rescue => e
+    @logger.error("Error registering plugin", :plugin => plugin.inspect, :error => e)
```

Comment:
ok but there are other places in the Pipeline code which also log `:error => e` ... if I correct this one then I should correct all the other too.